frontend/src/views/setting/index.vue

```diff
@@ -38,8 +36,4 @@ const buttons = [
         path: '/settings/about',
     },
 ];
-
-if (globalStore.isIntl) {
-    buttons.splice(4, 1);
-}
 </script>
```

Comment:
Your code has several improvements, but there are a few areas where you can optimize it further:

1. **Remove Unnecessary Variable Declaration**: The `globalStore` variable is only used once and assigned directly within the condition, so you can simplify this line to assign it on use.

2. **Use of Template Literals**: If you need to concatenate strings dynamically, using template literals is more concise and readable.

```typescript
import i18n from '@/lang';
// Removed: import { GlobalStore } from '@/store';

<script lang="ts" setup>
const globalStore: ReturnType<typeof import('@/store')> = await import('@/store').then(m => m.GlobalStore());

const buttons = [
    {
        text: 'Settings.',
        iconText: '<span aria-hidden="true">fas fa-cog</span>',
        path: '/settings/account',
    },
    ...[
        // other menu items remain unchanged...
    ],
];

if (globalStore.isIntl) {
    buttons.splice(4, 1); // Assuming the index is correct; adjust if necessary.
}
</script>
```

3. **Avoid Redundant Imports**: Ensure that all imports needed at the top level are included.

Here's the optimized code with these changes:

```typescript
import i18n from '@/lang';
import GlobalStore from '@/store';

const globalStore = new GlobalStore();

const buttons = [
    {
        text: 'Settings.',
        iconText: '<span aria-hidden="true">fas fa-cog</span>',
        path: '/settings/account',
    },
    // ... rest of button configurations ...
];

if (globalStore.isIntl) {
    buttons.splice(4, 1); // Adjust the index based on your logic needs.
}

<template>
    <!-- Your template content here -->
</template>

<script lang="ts">
export default defineComponent({
    components: {},
});
</script>

<style scoped>
/* Your styles go here */
</style>
```

By making these adjustments, your code will be cleaner, more efficient, and easier to maintain. Make sure to test thoroughly after applying these changes to ensure compatibility and functionality.